libyul/ObjectParser.cpp

```diff
@@ -104,10 +127,92 @@ shared_ptr<Block> ObjectParser::parseCode()
 	return parseBlock();
 }
 
+optional<ObjectParser::ReverseSourceNameMap> ObjectParser::tryGetSourceLocationMapping() const
+{
+	return tryGetSourceLocationMapping(m_scanner->currentCommentLiteral());
+}
+
+optional<ObjectParser::ReverseSourceNameMap> ObjectParser::tryGetSourceLocationMapping(std::string const& _text)
+{
+	// @use-src 0:"abc.sol" , "1:foo.sol" ,2:"bar.sol"
+	//
+	// UseSrcList := UseSrc (',' UseSrc)*
+	// UseSrc     := [0-9]+ ':' FileName
+	// FileName   := "(([^\"]|\.)*)"
+
+	// Matches some "@use-src TEXT".
+	static std::regex const lineRE = std::regex(
+		R"~~~((^|\s+)@use-src\s+(.*)$)~~~",
+		std::regex_constants::ECMAScript | std::regex_constants::optimize
+	);
+	string_view text(_text);
+	std::cmatch cm;
+	if (!std::regex_search(text.data(), text.data() + text.size(), cm, lineRE))
+		return nullopt;
+	solAssert(cm.size() == 3, "");
+
+	// Let @c text point to the parameter value (last match).
+	text = string_view(cm[2].first, static_cast<size_t>(std::distance(cm[2].first, cm[2].second)));
+
+	// iteratively match for NUM : STRING_LITERAL and increment
+	static regex const firstParamRE(R"~~~(\s*(\d+)\s*:\s*"((?:\\\"|[^\"])*)")~~~",
+		std::regex_constants::ECMAScript | std::regex_constants::optimize
+	);
+	static regex const continuationParamRE(R"~~~(\s*,\s*(\d+)\s*:\s*"((?:\\\"|[^\"])*)")~~~",
+		std::regex_constants::ECMAScript | std::regex_constants::optimize
+	);
+
+	ReverseSourceNameMap result;
+
+	int k = 0;
+	while (!text.empty())
+	{
+		if (!std::regex_search(text.data(), text.data() + _text.size(), cm, k ? continuationParamRE : firstParamRE))
+			return nullopt;
+		++k;
+		solAssert(cm.size() == 3, "");
+
+		auto const len = cm[0].length();
+		solAssert(len > 0, "");
+		text.remove_prefix(static_cast<size_t>(len));
+		solAssert(k <= 16, ""); // A sanity-check to avoid abuse. Should never happen.
```

Comment:
Why 16?

Reply:
16 was made up. But I wanted to add a limit in order to avoid excess or abuse. I can use any other arbitrary high number (256?).